### PR TITLE
Update openssl to 0.10

### PR DESCRIPTION
**Pull request by l4l**
openssl 0.9.24 have compilation issues on some platforms (e.g https://github.com/rust-lang/rustup.rs/issues/1518). This pr bums version and also have some optional formatting fixes at ceb365d

### CODE REVIEW

**Comment by humb1t** [comment]
@xitep please look at this beautiful PR :-)

File: src/error.rs [hunk 1]
```diff
@@ -239,91 +236,69 @@ impl Clone for Error {
...
-        &SslError::Ssl(ref es) => ErrorKind::Ssl(SslError::Ssl(es.clone())),
+    if let Some(io_err) = err.io_error() {
+        return ErrorKind::Io(clone_ioe(io_err));
+    }
+    if let Some(ssl_err) = err.ssl_error() {
+        return ErrorKind::Ssl(From::from(ssl_err.to_owned()));
     }
+
+    // xxx: return something as default?
+    unreachable!()
```
Comment (dead10ck) [inline src/error.rs]: Strange... this almost seems like a step back in terms of API soundness from openssl. How unfortunate.

**Comment by dead10ck** [comment]
This was merged in e574ce09de01d877032970f87574ac7365ee2b0b. Thanks!

**Comment by dead10ck** [comment]
This was just published in v0.8.0. Thanks again!

**Pull request closed without merging**